python/ray/serve/autoscaling_metrics.py

```diff
@@ -102,19 +139,30 @@ def window_average(
             The average of all the datapoints for the key on and after time
             window_start_timestamp_s, or None if there are no such points.
         """
-        datapoints = self.data[key]
-
-        idx = bisect.bisect(
-            a=datapoints,
-            x=TimeStampedValue(
-                timestamp=window_start_timestamp_s, value=0  # dummy value
-            ),
-        )
-        points_after_idx = datapoints[idx:]
+        points_after_idx = self._get_datapoints(key, window_start_timestamp_s)
 
         if do_compact:
             self.data[key] = points_after_idx
 
         if len(points_after_idx) == 0:
             return
         return sum(point.value for point in points_after_idx) / len(points_after_idx)
+
+    def max(self, key: str, window_start_timestamp_s: float):
+        """Perform a max operation for metric `key`
+
+        Args:
+            key(str): the metric name.
+            window_start_timestamp_s(float): the unix epoch timestamp for the
+              start of the window. The computed average will use all datapoints
+              from this timestamp until now.
+        Returns:
+            Max value of the data points for the key on and after time
+            window_start_timestamp_s, or None if there are no such points.
+        """
+        points_after_idx = self._get_datapoints(key, window_start_timestamp_s)
+
+        if len(points_after_idx) == 0:
+            return
+
+        return max(point.value for point in points_after_idx)
```

Comment:
```suggestion
        return max((point.value for point in points_after_idx), default=None)
```

[`max` takes in a default](https://docs.python.org/3/library/functions.html#max) in case the iterable is empty. With this, we can remove the above lines:

```
if len(points_after_idx) == 0:
    return
```

Reply:
following the window_average implementation, 

points_after_idx need to be iterable (None is not iterable)

Reply:
points_after_idx need to be iterable. I think that is why we have 
```
if len(points_after_idx) == 0:
    return
```

Reply:
Yeah, `points_after_idx` should be iterable. My suggestion is to replace the `if len(points_after_idx) == 0` check with the `default` flag in max.